extensions/vertx-graphql/deployment/src/main/java/io/quarkus/vertx/graphql/deployment/VertxGraphqlProcessor.java

```diff
@@ -18,4 +34,38 @@ FeatureBuildItem feature() {
     WebsocketSubProtocolsBuildItem websocketSubProtocols(CombinedIndexBuildItem combinedIndexBuildItem) {
         return new WebsocketSubProtocolsBuildItem("graphql-ws");
     }
+
+    @BuildStep
+    List<ReflectiveClassBuildItem> registerForReflection() {
+        return Arrays.asList(
+                new ReflectiveClassBuildItem(true, true,
+                        io.vertx.ext.web.handler.graphql.impl.GraphQLInputDeserializer.class.getName()),
+                new ReflectiveClassBuildItem(true, true, GraphQLBatch.class.getName()),
+                new ReflectiveClassBuildItem(true, true, GraphQLQuery.class.getName()));
+    }
+
+    @BuildStep
+    List<NativeImageResourceBuildItem> registerNativeImageResources() throws IOException {
+        ResolvedArtifact artifact = getVertxGraphQlArtifact();
+
+        List<NativeImageResourceBuildItem> nativeImageResourceBuildItems = new ArrayList<>();
+
+        File artifactFile = artifact.getArtifactPath().toFile();
+        try (JarFile jarFile = new JarFile(artifactFile)) {
+            Enumeration<JarEntry> entries = jarFile.entries();
+            while (entries.hasMoreElements()) {
+                JarEntry entry = entries.nextElement();
+                if (!entry.isDirectory() && entry.getName().startsWith("io/vertx/") && !entry.getName().endsWith(".class")) {
+                    nativeImageResourceBuildItems.add(new NativeImageResourceBuildItem(entry.getName()));
+                }
```

Comment:
Let's keep that one in mind in case we need it a second time, we could have a specific build item for it.